hail/python/hailtop/auth/auth.py

```diff
@@ -5,27 +5,31 @@
 from .tokens import get_tokens
 
 
-async def async_get_userinfo():
-    deploy_config = get_deploy_config()
-    headers = auth_headers('auth')
+async def async_get_userinfo(deploy_config=None):
+    if not deploy_config:
+        deploy_config = get_deploy_config()
+    headers = service_auth_headers(deploy_config, 'auth')
     async with aiohttp.ClientSession(
             raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
         async with session.get(
                 deploy_config.url('auth', '/api/v1alpha/userinfo'), headers=headers) as resp:
             return await resp.json()
 
 
-def get_userinfo():
-    return async_to_blocking(async_get_userinfo())
+def get_userinfo(deploy_config=None):
+    return async_to_blocking(async_get_userinfo(deploy_config))
 
 
-def auth_headers(service, authorize_target=True):
-    deploy_config = get_deploy_config()
+def namespace_auth_headers(deploy_config, ns, authorize_target=True):
     tokens = get_tokens()
-    ns = deploy_config.service_ns(service)
     headers = {}
     if authorize_target:
-        headers['Authorization'] = f'Bearer {tokens[ns]}'
+        headers['Authorization'] = f'Bearer {tokens.namespace_token_or_error(ns)}'
     if deploy_config.location() == 'external' and ns != 'default':
-        headers['X-Hail-Internal-Authorization'] = f'Bearer {tokens["default"]}'
+        headers['X-Hail-Internal-Authorization'] = f'Bearer {tokens.namespace_token_or_error("default")}'
     return headers
+
+
```

Comment:
Can you comment on where each of these methods should be used (what context)?

Reply:
These are used in all clients (hailctl, BatchClient, CI client, etc.) to set the headers necessarily to configure authorization.

Reply:
Let me know if you need more context.

Reply:
I meant who is the intended caller of namespace_auth_headers vs. service_auth_headers. The code all seems fine, so I can approve. Just trying to better understand what's going on.

Reply:
That's what I thought you were asking.  Anyone who is making calls to batch, ci, auth, notebook or other Hail service REST endpoints in our code will cause these functions to get the right authorization headers to include in the request.  The BatchClient constructor is a good example:

https://github.com/hail-is/hail/pull/7035/files#diff-b5b8ab69e6d64fe223a8d2c97dee5a60R453

It is calling service_auth_headers('batch') to get the headers for the client requests to the batch service.